dvc/path_info.py

```diff
@@ -148,12 +163,12 @@ def _fill_parts(self, scheme, host, user, port, path):
         self.port = int(port) if port else self.DEFAULT_PORTS.get(self.scheme)
 
         if isinstance(path, _URLPathInfo):
-            self._spath = str(path)
+            self.spath = str(path)
             self._path = path
```

Comment:
do we use `_spath` anywhere? is it effectively undefined in the `else`? what is the reason to rename it, btw (private to public)?

Reply:
I pressed "resolve conversation" before "comment". He is the explanation: Yes, it's used in `BaseOutput.get_file_name`.

Reply:
sorry, I had a loaded question (multiple points at once)

correct me if I'm wrong, your answer means that `spath` (no underscore `_`) is used outside, right? It makes sense then.

my first question was about `self._path = path` - do we still need `_spath`? And if we do, it looks like it might be undefined in certain cases? Is it expected?

Reply:
I didn't analyzed them deeply. I just made `spath` public. I didn't change the logics, so I don't know if it's expected `_path` being undefined in some cases. May I put a `TODO` comment here to investigate in another PR?

Reply:
Ah sorry, my bad, @julsal ! For some reason I mistakenly read `self._path = path` as `self._spath = path` and thought that we forgot to change it one place. No need to analyze this, it's not relevant at all. My mistake, sorry about the confusion.